src/background/stores/card-selection-store.ts

```diff
@@ -79,11 +81,11 @@ export class CardSelectionStore extends PersistentStore<CardSelectionStoreData>
     };
 
     private toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
-        if (!payload || !this.state.rules[payload.ruleId]) {
+        if (!payload || !this.state.rules || !this.state.rules[payload.ruleId]) {
             return;
         }
 
-        const rule = this.state.rules[payload.ruleId];
+        const rule = this.state.rules![payload.ruleId];
```

Comment:
You shouldn't need the `!` on `rules!` here (or any of the other usages in this file) - TypeScript is smart enough to narrow the typings based on the conditional checks you're making earlier in the function.

Reply:
Removed most instances, a few are still necessary to avoid strict null checks errors after we explicitly set it to a non-null value